backend/common/src/main/java/org/eclipse/sw360/cyclonedx/CycloneDxBOMImporter.java

```diff
@@ -987,16 +987,115 @@ public String getComponetNameById(String id, User user) throws SW360Exception {
      */
     public String sanitizeVCS(String vcs) {
         // GitHub repository URL Format: https://github.com/supplier/name
-        if (vcs.toLowerCase().contains("github.com")) {
-            URI uri = URI.create(vcs);
-            String[] urlParts = uri.getPath().split("/");
-            if (urlParts.length >= 3) {
-                String firstSegment = urlParts[1];
-                String secondSegment = urlParts[2].replaceAll("\\.git.*", "").replaceAll("#.*", "");
-                vcs = "https://github.com/" + firstSegment + "/" + secondSegment;
-                return vcs;
-            } else {
-                log.error("Invalid GitHub repository URL: " + vcs);
+        if (vcs.contains("github.com")) {
+            vcs = "https://" + vcs.substring(vcs.indexOf("github.com")).trim();
+            try {
+                URI uri = URI.create(vcs);
+                String[] urlParts = uri.getPath().split("/");
+
+                if (urlParts.length >= 3) {
+                    String firstSegment = urlParts[1];
+                    String secondSegment = urlParts[2].replaceAll("\\.git.*", "").replaceAll("#.*", "");
+                    vcs = "https://github.com/" + firstSegment + "/" + secondSegment;
+                    return vcs;
+                } else {
+                    log.error("Invalid GitHub repository URL: {}", vcs);
+                }
+            } catch (IllegalArgumentException e) {
+                log.error("Invalid URL format: {}", vcs, e);
+            }
+        }else if(vcs.contains("gitlab.com")){
+            // GitLab repository URL Format: https://gitlab.com/supplier/name
+            vcs = "https://" + vcs.substring(vcs.indexOf("gitlab.com")).trim();
+            try{
+                URI uri = URI.create(vcs);
+                String[] urlParts = uri.getPath().split("/");
+                if (urlParts.length >= 3) {
+                    String firstSegment = urlParts[1];
+                    String secondSegment = urlParts[2].replaceAll("\\.git.*", "").replaceAll("#.*", "");
+                    vcs = "https://gitlab.com/" + firstSegment + "/" + secondSegment;
```

Comment:
This will fail for repositories like https://gitlab.com/Orange-OpenSource/k8s-tz/applications/nexus
It will be changed to https://gitlab.com/Orange-OpenSource/k8s-tz which is technically wrong?

Reply:
Note, there can be more nested groups in GitLab. I just choose the first example I saw in the wild.